zipline/utils/calendars/calendar_helpers.py

```diff
@@ -0,0 +1,297 @@
+#
+# Copyright 2016 Quantopian, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pandas as pd
+import numpy as np
+import bisect
+from pytz import timezone
+
+from zipline.errors import NoFurtherDataError
+
+
+def normalize_date(date):
+    date = pd.Timestamp(date, tz='UTC')
+    return pd.tseries.tools.normalize_date(date)
+
+
+def delta_from_time(t):
+    """
+    Convert a datetime.time into a timedelta.
+    """
+    return pd.Timedelta(
+        hours=t.hour,
+        minutes=t.minute,
+        seconds=t.second,
+    )
+
+
+def _get_index(dt, all_trading_days):
+    """
+    Return the index of the given @dt, or the index of the preceding
+    trading day if the given dt is not in the trading calendar.
+    """
+    ndt = normalize_date(dt)
+    if ndt in all_trading_days:
+        return all_trading_days.searchsorted(ndt)
+    else:
+        return all_trading_days.searchsorted(ndt) - 1
+
+# The following methods are intended to be inserted in both the
+# ExchangeCalendar and TradingSchedule classes.
+# These methods live in the helpers module to avoid code duplication.
+
+
+def next_scheduled_day(date, last_trading_day, is_scheduled_day_hook):
+    """
+    Returns the next session date in the calendar after the provided date.
+
+    Parameters
+    ----------
+    date : Timestamp
+        The date whose following date is needed.
+
+    Returns
+    -------
+    Timestamp
+        The next scheduled date after the provided date.
+    """
+    dt = normalize_date(date)
```

Comment:
Or since we are storing the CustomBusinessDay.

```
def next_scheduled_day(self, date):
    dt = normalized_date(date)
    return dt + self.day
```
```